TokenCache becomes internal and only exposes TokenCachePersistenceOptions (#19240)
Azure/azure-sdk-for-net: sdk/identity/Azure.Identity/samples/ClientSideUserAuthentication.md

## Persisting user authentication data

Quite often applications desire the ability to be run multiple times without having to re-authenticate the user on each execution.

> This requires that data such a token or `AuthenticationRecord` from the original authentication be persisted outside of the application memory so that it can authenticate silently on subsequent executions.

Reviewer: ```suggestion
This requires that data from credentials be persisted outside of the application memory so that it can authenticate silently on subsequent executions.
```

> Various credential types can be configured to persist the required data by setting the `TokenPersistenceOptions`.

Reviewer: ```suggestion
Applications can persist this data using `TokenPersistenceOptions` when constructing the credential, and persisting the `AuthenticationRecord` returned from `Authenticate`.
```

> ### Persisting the TokenCache

Reviewer: ```suggestion
### Persisting the token cache
```

### Persisting the TokenCache

> The `TokenCache` contains all the data needed to silently authenticate, one or many accounts. It contains sensitive data such as refresh tokens, and access tokens and must be protected to prevent compromising the accounts it houses tokens for. The `Azure.Identity` library provides the `TokenCache` class which by default will protect and persist the cache using available platform data protection.

Reviewer: Since it's no longer a class we should probably reword the samples not to refer to `TokenCache` directly but talk about credentials caching tokens instead.

Author: I took a shot at rewording everything mentioning the `TokenCache`. Let me know how it looks.

}
```

> ### Silent authentication with AuthenticationRecord and TokenCache

Reviewer: ```suggestion
### Silent authentication with AuthenticationRecord and TokenCachePersistenceOptions
```
